File: src/main/java/project/rpg/listeners/InventoryEventListener.java
package project.rpg.listeners;

import org.bukkit.entity.Player;
import org.bukkit.event.EventHandler;
import org.bukkit.event.Listener;
import org.bukkit.event.inventory.InventoryClickEvent;
import org.bukkit.event.inventory.InventoryCloseEvent;
import project.rpg.base.GuiBase;

public class InventoryEventListener implements Listener {

    @EventHandler
    public void guiClick(InventoryClickEvent e){
        GuiBase gui = GuiBase.getGUI((Player) e.getWhoClicked());
        if(gui!=null) gui.onClick(e);
    }

    @EventHandler
    public void guiClose(InventoryCloseEvent e){
        GuiBase gui = GuiBase.getGUI((Player) e.getPlayer());
        if(gui!=null) gui.closeGUI(e);
    }

}

File: src/main/java/project/rpg/events/listeners/InventoryEventListener.kt
package project.rpg.events.listeners

import org.bukkit.event.EventHandler
import org.bukkit.event.Listener
import org.bukkit.event.inventory.*
import project.rpg.ui.inventory.GuiBase
import project.rpg.ui.inventory.interection.block.CraftingUI

class InventoryEventListener : Listener {
    @EventHandler
    fun guiClick(event: InventoryClickEvent) {
        CraftingUI.clickedBy(event)
        GuiBase.clickedBy(event)
    }

    @EventHandler
    fun guiInteraction(event: InventoryDragEvent) {
        CraftingUI.clickedBy(event)
    }
    @EventHandler
    fun guiInteraction(event: InventoryInteractEvent) {
        CraftingUI.clickedBy(event)
    }

    @EventHandler
    fun guiClose(event: InventoryCloseEvent) {
        GuiBase.closedBy(event)
        CraftingUI.closedBy(event)
    }

    @EventHandler
    fun craft(event:PrepareItemCraftEvent){

    }
}